Unity/Tests/BananaClient/TcpAdapterTest.cs

```diff
@@ -11,13 +10,97 @@
 using NUnit.Framework;
 using UnityEngine.TestTools;
 
-public class TcpAdapterTest : MonoBehaviour
+
+public class MemoryStreamWrapper: Stream
+{
+    private long readPos, writePos;
+
+    private bool _Disposed {get; set;} = false;
+
+    private readonly MemoryStream ms;
+
+    public override bool CanRead => ms.CanRead;
+
+    public override bool CanSeek => ms.CanSeek;
+
+    public override bool CanWrite => ms.CanWrite;
+
+    public override long Length => ms.Length;
+
+    public override long Position 
+    { 
+        get => ms.Position;
+        set => ms.Position = value; 
+    }
+
+    public override void Flush() => ms.Flush();
+
+    public override long Seek(long offset, SeekOrigin origin) => ms.Seek(offset, origin);
+
+    public override void SetLength(long value) => ms.SetLength(value);
+
+
+    public MemoryStreamWrapper()
+    {
+        readPos = writePos = 0;
+        ms = new MemoryStream();
+    }
+
+    public MemoryStreamWrapper(int capacity)
+    {
+        readPos = writePos = 0;
+        ms = new MemoryStream(capacity);
+    }
+
+    public override int Read(byte[] buffer, int offset, int count)
+    {
+        ms.Seek(readPos, SeekOrigin.Begin);
+        var byteRead = ms.Read(buffer, offset, count);
+        readPos = ms.Position;
+        //Debug.Log("ReadPos: " + readPos);
+        return byteRead;
+    }
+
+    public override int ReadByte()
+    {
+        ms.Seek(readPos, SeekOrigin.Begin);
+        var byteRead = ms.ReadByte();
+        readPos = ms.Position;
+        //Debug.Log("ReadPos: " + readPos);
+        return byteRead;
+    }
+
+    public override void Write(byte[] buffer, int offset, int count)
+    {
+        ms.Seek(writePos, SeekOrigin.Begin);
```

Comment:
Shouldn't it be seeking the end, instead of SeekOrigin.Begin?

Reply:
SeekOrigin.Begin is the relative position for the offset (first argument)

Reply:
i see. so `ms.Seek(writePos, SeekOrigin.Begin);` resets the offset to the current position which is the end of the stream?